### PR TITLE
Primitive type support for implicit parameters

### PR DESCRIPTION
**Pull request by ishaigor**
dataType="string" implicit params are failing without the changes. I added the tests and the fix that resolves the class by using io.swagger.util.ReflectionUtils

### CODE REVIEW

**Comment by ishaigor** [comment]
Need to rework

File: src/test/groovy/com/benjaminsproule/swagger/gradleplugin/MissingImplicitParamsITest.groovy [hunk 1]
```diff
@@ -30,7 +30,7 @@ class MissingImplicitParamsITest extends AbstractPluginITest {
 
         then:
         result.task(":${GenerateSwaggerDocsTask.TASK_NAME}").outcome == FAILED
-        result.output.contains("Caused by: java.lang.ClassNotFoundException: com.benjaminsproule.swagger.gradleplugin.test.model.MissingRequestModel")
+        result.output.contains("Unrecognized Type: [null]")
```
Comment (gigaSproule) [inline src/test/groovy/com/benjaminsproule/swagger/gradleplugin/MissingImplicitParamsITest.groovy]: This is a completely different and less helpful error message. It would be nice if instead of null it could print out the expected type.

**Comment by gigaSproule** [comment]
Is this required because string is defined in lowercase, whereas Java string is an object therefore uppercase first character? Don't get me wrong, if the changes help handle the edge cases, I'm happy with making these changes.

File: src/test/groovy/com/benjaminsproule/swagger/gradleplugin/MissingImplicitParamsITest.groovy (hunk 1 above)
Comment (ishaigor) [reply src/test/groovy/com/benjaminsproule/swagger/gradleplugin/MissingImplicitParamsITest.groovy]: This is the result of using io.swagger.util.ReflectionUtils#typeFromString. If you can tell me where it comes from - I can try and fix it.

**Comment by ishaigor** [comment]
> Is this required because string is defined in lowercase, whereas Java string is an object therefore uppercase first character? Don't get me wrong, if the changes help handle the edge cases, I'm happy with making these changes.

This is the same dataTypes that ApiModelProperty already supports: http://docs.swagger.io/swagger-core/v1.5.X/apidocs/io/swagger/annotations/ApiModelProperty.html#dataType-- and http://docs.swagger.io/swagger-core/v1.5.X/apidocs/io/swagger/annotations/ApiImplicitParam.html#dataType--

> The data type of the parameter.

> This can be the class name or a primitive.

The code that handles the implicit param (io.swagger.jaxrs.Reader#readImplicitParam) uses io.swagger.util.ReflectionUtils#typeFromString. The code that generates the  swagger for ApiImplicitParam could too. I can back out the change and contribute the test cases that show the bug only.
